src/content/tools/pub/dependencies.md

```diff
@@ -151,6 +151,12 @@ dependencies:
     git: git@github.com:munificent/kittens.git
 ```
 
+Alternatively, you can access private repos using access tokens via HTTPS.
```

Comment:
It might be better to be more generic.

--------

If you have
```yaml
dependencies:
  kittens:
    git: <url>
```

Then this will work, so long as `git clone <url>` will work.
 * If your `<url>` uses SSH (example: `git@github.com:munificent/kittens.git`), you may need to [configure an SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).
 * If your `<url>` uses HTTPS (example: `https://github.com/munificent/kittens.git`), you may need to configure [git credentials](https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git)).

Neither of these choices is specific to `dart pub`, which merely runs `git` as a subprocess.

--------

That's probably not what you want to write. My point is that the hint we should give here is probably not that "you can use HTTP to avoid continuously generating SSH keys". Because SSH keys should not be continuously generated!

Normally you generate an SSH key on your machine and copy the public key into your settings on github (or whatever git hosting provider you have).

Instead, I propose that the hint we give is that SSH isn't the only way to access private git repositories. Indeed, `dart pub` is not involved with the authentication when cloning a git repository.

I suppose that by trying to help and give example, we've potentially failed to clearly communicate that `dart pub` just calls `git clone` as a subprocess. And it is your responsibility to:
 * (a) configure `git` such that it can clone the repository your dependency is in.
   This **may** involve:
   * Creating a public/private SSH key-pair on your system, and uploading it to Github.
   * Installing and using `gh` to configure HTTPS access for Github.
   * Using `gcloud init` to configure a git-authentication helper for Cloud Source Repositories.
   * Using gpg to generate a key-pair on a yubikey and configuring `gpg-agent` (I've personally done this, it's rather complicated :rofl)
   * Using a u2f-key (yubikey) to create an SSH-key key-pair with `ssh-agent`-
   * etc... the list goes on and we should probably only provide hints for people using github.
 * (b) provide a `<url>` that works when `git clone <url>` is executed.

So describe the comment being very long -- perhaps what I'm trying to say is:
We should say less, or make it clear that using SSH is just an example, not the only way.

Reply:
💯  comment. I think we should just communicate the end of this comment with something like "`dart pub` calls `git clone` as a subprocess, so you need to provide a url that works when `git clone <url>` is executed. This is most commonly done with HTTPS access to Github or with an SSH key-pair on your system that you've uploaded to Github." 

And then link the second sentence to the [Github authentication page](https://docs.github.com/en/authentication).

I don't think you need to exhaustively discuss all the ways one might connect to Github

Reply:
Thank you both for the feedback! I changed it to kind of equate SSH and HTTPS just as different options to the same end, and then included the line about conforming to `git clone`